phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -4168,4 +4176,197 @@ public MutationState useSchema(UseSchemaStatement useSchemaStatement) throws SQL
+        }
+        logger.info(revokePermLog.toString());
+
+        HConnection hConnection = connection.getQueryServices().getAdmin().getConnection();
+
+        try {
+            if (revokeStatement.getSchemaName() != null) {
+                // SYSTEM.CATALOG doesn't have any entry for "default" HBase namespace, hence we will bypass the check
+                if(!revokeStatement.getSchemaName().equals(QueryConstants.HBASE_DEFAULT_SCHEMA_NAME)) {
+                    FromCompiler.getResolverForSchema(revokeStatement.getSchemaName(), connection);
+                }
+                revokePermissionsFromSchema(hConnection, revokeStatement);
+
+            } else if (revokeStatement.getTableName() != null) {
+                PTable inputTable = PhoenixRuntime.getTable(connection,
+                        SchemaUtil.normalizeFullTableName(revokeStatement.getTableName().toString()));
+                if (!(PTableType.TABLE.equals(inputTable.getType()) || PTableType.SYSTEM.equals(inputTable.getType()))) {
+                    throw new AccessDeniedException("Cannot REVOKE permissions from INDEX TABLES or VIEWS");
+                }
+                revokePermissionsFromTables(hConnection, revokeStatement, inputTable);
+
+            } else {
+                revokePermissionsFromUser(hConnection, revokeStatement);
+            }
+
+        } catch (SQLException e) {
+            // Bubble up the SQL Exception
+            throw e;
+        } catch (Throwable throwable) {
+            // Wrap around other exceptions to PhoenixIOException (Ex: org.apache.hadoop.hbase.security.AccessDeniedException)
+            throw ServerUtil.parseServerException(throwable);
+        }
+
+        return new MutationState(0, 0, connection);
+    }
+
+
+    private void revokePermissionsFromTables(HConnection hConnection, RevokeStatement revokeStatement, PTable inputTable) throws Throwable {
+
+        org.apache.hadoop.hbase.TableName tableName = SchemaUtil.getPhysicalTableName
+                (inputTable.getName().getBytes(), inputTable.isNamespaceMapped());
+
+        revokePermissionsFromTable(hConnection, revokeStatement, tableName);
+
+        for(PTable indexTable : inputTable.getIndexes()) {
+            // Local Indexes don't correspond to new physical table, they are just stored in separate CF of base table.
+            if(indexTable.getIndexType().equals(IndexType.LOCAL)) {
+                continue;
+            }
+            logger.info("Revoking perms from IndexTable: " + indexTable.getName() + " BaseTable: " + inputTable.getName());
+            if (inputTable.isNamespaceMapped() != indexTable.isNamespaceMapped()) {
+                throw new TablesNotInSyncException(inputTable.getTableName().getString(),
+                        indexTable.getTableName().getString(), "Namespace properties");
+            }
+            tableName = SchemaUtil.getPhysicalTableName(indexTable.getName().getBytes(), indexTable.isNamespaceMapped());
```

Comment:
you could use indexTable.getPhysicalName api instead.

Reply:
Sure. Will refactor.

Reply:
We need HBase TableName object, whereas getPhysicalName api returns PName, so it wont help us here.

Reply:
This doesn't look correct, @karanmehta93. You need to use indexTableName.getPhysicalName().getBytes() here *and* then call SchemaUtil.getPhysicalTableName like this:

    tableName = SchemaUtil.getPhysicalTableName(indexTableName.getPhysicalName().getBytes(), indexTable.isNamespaceMapped());

Otherwise, for something like an index on a view or a local index, you wouldn't get a physical table name.

Reply:
The code that I wrote also works because it takes in the original name and uses its namespace mapped property to determine the Physical table name. 
If you see the code for `getPhysicalName()` in `PTableImpl`, it is the `SchemaUtil.getPhysicalHBaseTableName(schemaName, tableName, isNamespaceMapped)`
The simplest and cleaner option in this case is to use `TableName.valueOf(indexTableName.getPhysicalName().getBytes())`. 
I was trying to avoid using the HBase API's directly and use the SchemaUtil methods. What do you suggest?
@JamesRTaylor